app/assets/javascripts/map.js

```diff
@@ -125,25 +207,210 @@ class Map {
         position: results[0].geometry.location
       });
       this.geocodedAddress = results[0].formatted_address;
-      this.onGeocodingResponse(this.geocodedAddress);
+      this.updateAddressOnSummary(this.geocodedAddress);
     } else {
       alert('Geocode was not successful for the following reason: ' + status);
     }
   }
 
+  updateAddressOnSummary(geocodedAddress) {
+    if (geocodedAddress) {
+      this.geocodedAddress = geocodedAddress;
+      const primaryAddressNode = document.getElementById('primaryAddress');
+      const secondaryAddressNode = document.getElementById('secondaryAddress');
+      while (primaryAddressNode.firstChild) { primaryAddressNode.removeChild(primaryAddressNode.firstChild) }
+      while (secondaryAddressNode.firstChild) { secondaryAddressNode.removeChild(secondaryAddressNode.firstChild) }
+      
+      let splitAddress = geocodedAddress.split(',');
+      const div = document.createElement('div');
+      div.innerText = splitAddress.shift();
+      div.classList.add('primary-address');
+      primaryAddressNode.appendChild(div);
+      splitAddress.forEach(element => {
+        const div = document.createElement('div');
+        div.innerText = element;
+        div.classList.add('secondary-address');
+        secondaryAddressNode.appendChild(div);
+      });
+    Dom.showNode(document.getElementById('summary'));
+    Dom.showNode(document.getElementById('summaryAddress'));
+    }
+  }
+
   handlePolygonCreated(polygon){
     this.polygons.push(polygon)
-    this.onPolygonsChanged(this.polygons);
+    this.handlePolygonChanged();
   }
 
   handleRemoveLastPolygon() {
     this.polygons.pop().setMap(null);
-    this.onPolygonsChanged(this.polygons);
+    this.handlePolygonChanged(this.polygons);
   }
 
   handleRemoveAllPolygons() {
     this.polygons.forEach(p => p.setMap(null));
     this.polygons = [];
-    this.onPolygonsChanged(this.polygons);
+    this.handlePolygonChanged(this.polygons);
+  }
+
+  handleDoneSelecting() {
+    this.showNextSection('collapseMap', 'collapseAddOns');
```

Comment:
You could probably "automate" this without having to specify the current and the next. You could _find_ which is the actively open one, and then find it's first _sibling_ all within `showNextSection` (no arguments)